checkov/bitbucket_pipelines/runner.py

```diff
@@ -0,0 +1,135 @@
+import os
+
+from checkov.bitbucket_pipelines.checks.registry import registry
+from checkov.common.images.image_referencer import ImageReferencer, Image
+from checkov.common.output.report import CheckType
+from checkov.yaml_doc.runner import Runner as YamlRunner
+
+
+class Runner(YamlRunner, ImageReferencer):
+    check_type = CheckType.BITBUCKET_PIPELINES
+
+    def __init__(self):
+        super().__init__()
+
+    def require_external_checks(self):
+        return False
+
+    def import_registry(self):
+        return registry
+
+    def _parse_file(self, f):
+        if self.is_workflow_file(f):
+            return super()._parse_file(f)
+
+    def is_workflow_file(self, file_path):
+        """
+        :return: True if the file mentioned is in a is named bitbucket-pipelines.yml. Otherwise: False
+        """
+        abspath = os.path.abspath(file_path)
+        return abspath.endswith(("bitbucket-pipelines.yml", "bitbucket-pipelines.yaml"))
+
+    def get_images(self, file_path):
+        """
+        Get container images mentioned in a file
+        :param file_path: File to be inspected
+
+        File sample that will return 4 Image objects:
+        #image: node:10.15.0
+        #
+        #pipelines:
+        #  default:
+        #    - step:
+        #        name: Build and test
+        #        image: node:10.0.0
+        #        script:
+        #          - npm install
+        #          - npm test
+        #          - npm run build
+        #        artifacts:
+        #          - dist/**
+        #    - step:
+        #        name: Deploy
+        #        image: python:3.7.2
+        #        trigger: manual
+        #        script:
+        #          - python deploy.py
+        #  custom:
+        #    sonar:
+        #      - step:
+        #          image: python:3.8.2
+        #          script:
+        #            - echo "Manual triggers for Sonar are awesome!"
+        #    deployment-to-prod:
+        #      - step:
+        #          script:
+        #            - echo "Manual triggers for deployments are awesome!"
+        #  branches:
+        #    staging:
+        #      - step:
+        #          script:
+        #            - echo "Auto pipelines are cool too."
+        :return: List of container image objects mentioned in the file.
+
+        """
+
+        images = set()
+
+        workflow, workflow_line_numbers = self._parse_file(file_path)
+        root_image = workflow.get("image", "")
+        self.add_root_image(file_path, images, root_image, workflow_line_numbers)
+
+        pipelines = workflow.get("pipelines", {})
+        default_pipeline = pipelines.pop("default", {})
+        self.add_default_pipeline_images(file_path, images, default_pipeline)
+
+        self.add_pipeline_images(file_path, images, pipelines)
+
+        return images
+
+    def add_root_image(self, file_path: str, images: [], root_image: str, workflow_line_numbers: dict) -> None:
```

Comment:
```suggestion
    def add_root_image(self, file_path: str, images: list, root_image: str, workflow_line_numbers: tuple[int, str]) -> None:
```
I also recommend to add this line to the top of the file, then you can use the built-in types for type hints
```python
from __future__ import annotations
```